shared-bindings/displayio/Bitmap.c

```diff
@@ -172,6 +172,115 @@ STATIC mp_obj_t bitmap_subscr(mp_obj_t self_in, mp_obj_t index_obj, mp_obj_t val
     return mp_const_none;
 }
 
+//|     def blit(self, x: int, y: int, source_bitmap: bitmap, x1: int, y1: int, x2: int, y2: int, skip_index: int) -> Any:
+//|         """Inserts the source_bitmap region defined by rectangular boundaries
+//|         (x1,y1) and (x2,y2) into the bitmap at the specified (x,y) location.
+//|         :param int x: Horizontal pixel location in bitmap where source_bitmap upper-left
+//|             corner will be placed
+//|         :param int y: Vertical pixel location in bitmap where source_bitmap upper-left
+//|             corner will be placed
+//|         :param bitmap source_bitmap: Source bitmap that contains the graphical region to be copied
+//|         : param int x1: Minimum x-value for rectangular bounding box to be copied from the source bitmap
+//|         : param int y1: Minimum y-value for rectangular bounding box to be copied from the source bitmap
+//|         : param int x2: Maximum x-value for rectangular bounding box to be copied from the source bitmap
+//|         : param int y2: Maximum y-value for rectangular bounding box to be copied from the source bitmap
+//|         : param int skip_index: bitmap palette index in the source that will not be copied, set `None` to copy all pixels
+//|         ...
+//|
+
+STATIC mp_obj_t displayio_bitmap_obj_blit(size_t n_args, const mp_obj_t *pos_args, mp_map_t *kw_args){
+    enum {ARG_x, ARG_y, ARG_source, ARG_x1, ARG_y1, ARG_x2, ARG_y2, ARG_skip_index};
+    static const mp_arg_t allowed_args[] = {
+        {MP_QSTR_x, MP_ARG_REQUIRED | MP_ARG_INT},
+        {MP_QSTR_y, MP_ARG_REQUIRED | MP_ARG_INT},
+        {MP_QSTR_source, MP_ARG_REQUIRED | MP_ARG_OBJ},
+        {MP_QSTR_x1, MP_ARG_KW_ONLY | MP_ARG_OBJ, {.u_obj = mp_const_none} },
+        {MP_QSTR_y1, MP_ARG_KW_ONLY | MP_ARG_OBJ, {.u_obj = mp_const_none} },
+        {MP_QSTR_x2, MP_ARG_KW_ONLY | MP_ARG_OBJ, {.u_obj = mp_const_none} }, // None convert to source->width
+        {MP_QSTR_y2, MP_ARG_KW_ONLY | MP_ARG_OBJ, {.u_obj = mp_const_none} }, // None convert to source->height
+        {MP_QSTR_skip_index, MP_ARG_OBJ | MP_ARG_KW_ONLY, {.u_obj=mp_const_none} },
+    };
+    mp_arg_val_t args[MP_ARRAY_SIZE(allowed_args)];
+    mp_arg_parse_all(n_args - 1, pos_args + 1, kw_args, MP_ARRAY_SIZE(allowed_args), allowed_args, args);
+
+    displayio_bitmap_t *self = MP_OBJ_TO_PTR(pos_args[0]); //*******
+
+    int16_t x = args[ARG_x].u_int;
+    int16_t y = args[ARG_y].u_int;
+
+    displayio_bitmap_t *source = MP_OBJ_TO_PTR(args[ARG_source].u_obj);
+
+    int16_t x1, y1;
+    // if x1 or y1 is None, then set as the zero-point of the bitmap
+    if ( args[ARG_x1].u_obj == mp_const_none ) {
+        x1 = 0;
+    } else {
+        x1 = mp_obj_get_int(args[ARG_x1].u_obj);
+    }
+    //int16_t y1;
+    if ( args[ARG_y1].u_obj == mp_const_none ) {
+        y1 = 0;
+    } else {
+        y1 = mp_obj_get_int(args[ARG_y1].u_obj);
+    }
```

Comment:
These can be removed by setting the arg array above to `MP_ARG_INT` with `.u_int = 0`.

Reply:
Updated per your request.  (Somehow I got hung up on the x2,y2 checking and decided this needed it too.)